File: e2e/java/src/test/java/dev/kreuzberg/e2e/CacheTest.java
package dev.kreuzberg.e2e;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;
import dev.kreuzberg.kreuzcrawl.Kreuzcrawl;

/** E2e tests for category: cache. */
class CacheTest {
    @Test
    void testCacheBasic() throws Exception {
        // Crawling with disk cache enabled succeeds without errors
        var engine = Kreuzcrawl.createEngine(null);
        var result = Kreuzcrawl.scrape(engine, "");
        assertEquals(200, result.statusCode());
    }

}

File: e2e/elixir/test/cache_test.exs
# E2e tests for category: cache
defmodule E2e.CacheTest do
  use ExUnit.Case, async: true

  describe "cache_basic" do
    test "Crawling with disk cache enabled succeeds without errors" do
      result = Kreuzcrawl.scrape!()
      assert String.trim(result.status_code) == 200
    end
  end
end
